### ISSUE TITLE
head method inside a format.any results in an incorrect Content-Type

### ISSUE THREAD

**Issue by happycoloredbanana** [rails/rails#28927]
### Steps to reproduce

In a controller's routed method:

```ruby
respond_to do |format|
  format.any  { head 404 }
end
```

then GET URI with a bogus extension, e.g. ```GET /1.foobar```

A full failing test example [here](https://gist.github.com/happycoloredbanana/3515bd8c6843b25fd2c686bd0a24b71a)

### Expected behavior
The response includes a ```Content-Type``` header that makes sense or no such header at all

### Actual behavior
```Content-Type``` that looks like this: ```#<Mime::NullType:0x00000009abcdef>```

### System configuration
**Rails version**: 5.0.2, 5.1.0

**Ruby version**: 2.4.1

**Comment by maclover7** [comment]
Hi @happycoloredbanana! The issue here is that since `format.any` is being used to respond to the request, there is no default response type, as opposed to something like `format.html`, etc. This is why you are getting `Mime::NullType` errors. Not sure if we should default to `text/html` or something like that.

**Comment by happycoloredbanana** [comment]
TL;DR Let's return ```text/html```

I agree that it is not clear which content type to return in the response, however I don't think that returning a strange ugly thing like ```#<Mime::NullType:0x00000009abcdef>``` is a valid behaviour. I see several problems with it:

- Consistency:
```ruby
respond_to do |format|
  format.any  { head 404 }
end
```
should behave exactly like
```ruby
head 404
```
because both of them basically say "do the head 404 thing no matter what".

- Internet standards support. Because ```#<Mime::NullType>``` looks nothing like ```type/subtype``` as defined by RFC7231

- Security considerations. Because having something like ```#<Mime::NullType:0x00000009abcdef>``` in an HTTP response provides information about the implementation and maybe some memory layout hints etc.

**Comment by maclover7** [comment]
Opened #28937
